vaporetto/src/predictor.rs

```diff
@@ -1,1136 +1,861 @@
+                        vec!["ジン".into(), "ヒト".into()],
+                    ],
+                    char_ngram_model: TagNgramModel(vec![TagNgramData {
+                        ngram: "は地球人".into(),
+                        weights: vec![(0, vec![-32, 33, 34, -35])],
+                    }]),
+                    type_ngram_model: TagNgramModel(vec![TagNgramData {
+                        ngram: vec![Hiragana as u8, Kanji as u8, Hiragana as u8],
+                        weights: vec![(1, vec![36, -37, -38, 39])],
+                    }]),
+                    bias: vec![40, 41, 42, 43],
+                },
+                TagModel {
+                    token: "地球".into(),
+                    tags: vec![
+                        vec!["名詞".into()],
+                        vec!["マンホーム".into(), "チキュー".into()],
+                    ],
+                    char_ngram_model: TagNgramModel(vec![TagNgramData {
+                        ngram: "は地球人".into(),
+                        weights: vec![(1, vec![-44, 45])],
+                    }]),
+                    type_ngram_model: TagNgramModel(vec![]),
+                    bias: vec![46, 47],
+                },
             ],
-            s.boundaries(),
-        );
+        )
     }
 
     #[test]
-    fn test_predict_with_score_4() {
-        let model = generate_model_4();
-        let p = Predictor::new(model, false).unwrap();
-        let s = Sentence::from_raw("我らは全世界の国民").unwrap();
-        let s = p.predict_with_score(s);
-        assert_eq!(&[-77, 38, 89, 219, 221, 233, 94, 12], s.boundary_scores(),);
+    fn test_predict_boundaries() {
+        let model = create_test_model();
+        let predictor = Predictor::new(model, false).unwrap();
+        let mut sentence = Sentence::from_raw("この人は地球人だ").unwrap();
+        predictor.predict(&mut sentence);
+        assert_eq!(&[-22, 54, 58, 43, -54, 68, 48], sentence.boundary_scores(),);
         assert_eq!(
             &[
-                BoundaryType::NotWordBoundary,
-                BoundaryType::WordBoundary,
-                BoundaryType::WordBoundary,
-                BoundaryType::WordBoundary,
-                BoundaryType::WordBoundary,
-                BoundaryType::WordBoundary,
-                BoundaryType::WordBoundary,
-                BoundaryType::WordBoundary,
+                NotWordBoundary,
+                WordBoundary,
+                WordBoundary,
+                WordBoundary,
+                NotWordBoundary,
+                WordBoundary,
+                WordBoundary
             ],
-            s.boundaries(),
+            sentence.boundaries(),
         );
     }
 
     #[cfg(feature = "tag-prediction")]
     #[test]
-    fn test_predict_with_score_5() {
-        let model = generate_model_5();
-        let p = Predictor::new(model, true).unwrap();
-        let s = Sentence::from_raw("人と人をつなぐ人").unwrap();
-        let mut s = p.predict(s);
+    fn test_predict_tags() {
```

Comment:
Could you add another test to check the case that `sentence.fill_tags()` is called when a predictor without tags is set?

Reply:
d993af3